pkg/comp-functions/functions/vshnminio/minio_deploy.go

```diff
@@ -0,0 +1,275 @@
+			Labels: map[string]string{
+				"appcat.vshn.io/servicename":     "minio-distributed",
+				"appcat.vshn.io/claim-namespace": comp.GetClaimNamespace(),
+				"appuio.io/no-rbac-creation":     "true",
+				"appuio.io/billing-name":         "appcat-minio"},
+		},
+	}
+
+	return iof.Desired.PutIntoObject(ctx, ns, comp.Name+"-ns")
+}
+
+// Create the helm release for the minio instance
+func createObjectHelmRelease(ctx context.Context, comp *vshnv1.VSHNMinio, iof *runtime.Runtime) error {
+
+	plan := GetPlan(comp, iof)
+
+	resouces, err := common.FetchPlansFromConfig(ctx, iof, plan)
+	if err != nil {
+		return err
+	}
+
+	reqMem := ""
+	reqCPU := ""
+	mem := ""
+	cpu := ""
+	disk := ""
+
+	if comp.Spec.Parameters.Size.Requests.Memory != "" {
+		reqMem = comp.Spec.Parameters.Size.Requests.Memory
+	} else {
+		reqMem = resouces.MemoryRequests.String()
+	}
+	if comp.Spec.Parameters.Size.Requests.CPU != "" {
+		reqCPU = comp.Spec.Parameters.Size.Requests.CPU
+	} else {
+		reqCPU = resouces.CPURequests.String()
+	}
+	if comp.Spec.Parameters.Size.Memory != "" {
+		mem = comp.Spec.Parameters.Size.Memory
+	} else {
+		mem = resouces.MemoryLimits.String()
+	}
+	if comp.Spec.Parameters.Size.CPU != "" {
+		cpu = comp.Spec.Parameters.Size.CPU
+	} else {
+		cpu = resouces.CPULimits.String()
+	}
+	if comp.Spec.Parameters.Size.Disk != "" {
+		disk = comp.Spec.Parameters.Size.Disk
+	} else {
+		disk = resouces.Disk.String()
+	}
+
+	values := map[string]interface{}{
+		"replicas": comp.Spec.Parameters.Instances,
+		"resources": map[string]interface{}{
+			"requests": map[string]interface{}{
+				"memory": reqMem,
+				"cpu":    reqCPU,
+			},
+			"limits": map[string]interface{}{
+				"memory": mem,
+				"cpu":    cpu,
+			},
+		},
+		"persistence": map[string]interface{}{
+			"size": disk,
+		},
+	}
+
+	vb, err := json.Marshal(values)
+	if err != nil {
+		return err
+	}
+
+	r := &xhelmbeta1.Release{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: comp.GetName(),
+		},
+		Spec: xhelmbeta1.ReleaseSpec{
+			ForProvider: xhelmbeta1.ReleaseParameters{
+				Chart: xhelmbeta1.ChartSpec{
+					Repository: iof.Config.Data["minioChartRepository"],
+					Version:    iof.Config.Data["minioChartVersion"],
+					Name:       "minio",
+				},
+				Namespace: comp.GetInstanceNamespace(),
+				ValuesSpec: xhelmbeta1.ValuesSpec{
+					Values: k8sruntime.RawExtension{
+						Raw: vb,
+					},
+				},
+			},
+			ResourceSpec: xpv1.ResourceSpec{
+				ProviderConfigReference: &xpv1.Reference{
+					Name: "helm",
+				},
+			},
+		},
+	}
+
+	return iof.Desired.PutIntoObject(ctx, r, comp.Name+"-release")
+}
+
+func createSecretObserver(ctx context.Context, comp *vshnv1.VSHNMinio, iof *runtime.Runtime) error {
+
+	secret := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-minio",
+			Namespace: comp.GetInstanceNamespace(),
+		},
+	}
+
+	connection := &xkubev1.Object{
```

Comment:
Instead of building the `xkubev1.Object` here you could use the `iof.PutIntoObserveOnlyObject` function.

Reply:
Fixed